benchmark CoreMark

ingallsj opened the pull request:
copy and edit only the necessary files from https://github.com/eembc/coremark commit 509d8ef on May 31

ingallsj committed:
add Core-Mark to benchmarks

aswaterman commented:
Has EEMBC changed the license for coremark? In 2016, I checked and we were not allowed to distribute it.

ingallsj commented:
> Has EEMBC changed the license for coremark? In 2016, I checked and we were not allowed to distribute it.

I'm not sure, and my hesitation on this point led me to push this into sifive/riscv-tests rather than riscv/riscv-tests, for what that's worth.

From https://www.eembc.org/coremark/download.php:
"CoreMark is now available to license free of charge on GitHub! All previous licenses have been superseded by the Apache license and the Acceptible Usage disclosure, but please read and be aware of these documents found in the repository."

aswaterman commented:
Ah, it’s nice that it’s now Apache.

jim-wilson commented:
The license has changed, but there is a lot of verbiage to protect their trademarks, which we need to be careful about.  There is incidentally an outstanding request from Arun Thomas to add it to freedom-e-sdk.

ingallsj commented:
Should I keep only the files that I changed from the main CoreMark repo, and remove the un-modified files?

Changed: core_portme.h, core_portme.c

Identical: core_list_join.c, core_main.c, core_matrix.c, core_state.c, core_util.c, coremark.h
